src/adapters/adkernel.js

```diff
@@ -0,0 +1,194 @@
+var bidmanager = require('../bidmanager.js');
+var bidfactory = require('../bidfactory.js');
+var ajax = require('../ajax.js');
+var utils = require('../utils.js');
+
+/**
+ * Adapter for requesting bids from AdKernel white-label platform
+ * @class
+ */
+var AdkernelAdapter = function AdkernelAdapter() {
+  const ADKERNEL = 'adkernel';
+
+  /**
+   * Helper object to build multiple bid requests in case of multiple zones/ad-networks
+   * @constructor
+   */
+  function RtbRequestDispatcher() {
+    const imp2adUnit = {},
+      _dispatch = {},
+      site = createSite();
+
+    //translate adunit info into rtb impression dispatched by host/zone
+    this.addImp = function (host, zone, width, height, bidId, adUnitId) {
+      if (!(host in _dispatch)) {
+        _dispatch[host] = {};
```

Comment:
If `host` is undefined, you should abort this request. Without the host param, I was seeing network requests to `http://undefined/....`